includes/reader-revenue/class-stripe-connection.php

```diff
@@ -438,55 +438,66 @@ public static function receive_webhook( $request ) {
-							$contact['metadata']
+							$contact['metadata'],
+							[
+								Newspack_Newsletters::$metadata_keys['last_payment_date']   => $payment_date,
+								Newspack_Newsletters::$metadata_keys['last_payment_amount'] => $amount_normalised,
+							]
 						);
+
+						if ( 'once' === $frequency ) {
+							$contact['metadata'][ Newspack_Newsletters::$metadata_keys['membership_status'] ] = 'Donor';
+						} else {
+							$contact['metadata'] = array_merge(
+								self::create_recurring_payment_metadata( $frequency, $payment['amount'], $payment['currency'], $payment['created'] ),
+								$contact['metadata']
+							);
+						}
+						if ( isset( $customer['metadata']['userId'] ) ) {
+							$contact['metadata'][ Newspack_Newsletters::$metadata_keys['account'] ] = $customer['metadata']['userId'];
+						}
+						if ( isset( $customer['metadata']['current_page_url'] ) ) {
+							$contact['metadata']['current_page_url'] = $customer['metadata']['current_page_url'];
+						}
+
+						if ( Donations::is_woocommerce_suite_active() ) {
+							$wc_product_id = Donations::get_donation_product( $frequency );
+							try {
+								$wc_product = \wc_get_product( $wc_product_id );
+								$contact['metadata'][ Newspack_Newsletters::$metadata_keys['product_name'] ] = $wc_product->get_name();
+							} catch ( \Throwable $th ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
+								// Fail silently.
+							}
+						}
 					}
 
 					if ( ! empty( $client_id ) ) {
 						$contact['client_id'] = $client_id;
 					}
-					if ( isset( $customer['metadata']['userId'] ) ) {
-						$contact['metadata'][ Newspack_Newsletters::$metadata_keys['account'] ] = $customer['metadata']['userId'];
-					}
-					if ( isset( $customer['metadata']['current_page_url'] ) ) {
-						$contact['metadata']['current_page_url'] = $customer['metadata']['current_page_url'];
-					}
 
-					if ( Donations::is_woocommerce_suite_active() ) {
-						$wc_product_id = Donations::get_donation_product( $frequency );
-						try {
-							$wc_product = \wc_get_product( $wc_product_id );
-							$contact['metadata'][ Newspack_Newsletters::$metadata_keys['product_name'] ] = $wc_product->get_name();
-						} catch ( \Throwable $th ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
-							// Fail silently.
-						}
-					}
-
-					if ( method_exists( '\Newspack_Newsletters_Subscription', 'add_contact' ) ) {
-						// Note: With Mailchimp, this is adding the contact as 'pending' - the subscriber has to confirm.
-						if ( ! empty( $stripe_data['newsletter_list_id'] ) && $has_opted_in_to_newsletters ) {
-							\Newspack_Newsletters_Subscription::add_contact( $contact, $stripe_data['newsletter_list_id'] );
-						} else {
-							\Newspack_Newsletters_Subscription::add_contact( $contact );
-						}
-						$was_customer_added_to_mailing_list = true;
+					// Note: With Mailchimp, this is adding the contact as 'pending' - the subscriber has to confirm.
```

Comment:
The contact is not added as 'pending' to Mailchimp

Reply:
This was not added in this PR (just moved lines) – can you add an issue?